node/src/main/scala/com/wavesplatform/state/reader/CompositeBlockchain.scala

```diff
@@ -95,7 +95,7 @@ final class CompositeBlockchain private (
     }
 
   override def balanceSnapshots(address: Address, from: Int, to: Option[BlockId]): Seq[BalanceSnapshot] =
-    if (maybeDiff.isEmpty || to.exists(id => inner.heightOf(id).isDefined)) {
+    if (maybeDiff.isEmpty || to.exists(id => inner.heightOf(id).exists(_ != height))) {
```

Comment:
Could you add a comment here, why do we need to check the `height`?

Reply:
The height of the composite blockchain can match the height of the nested blockchain if there is no block in the composite one.